core/station/impl/src/models/request_policy.rs

```diff
@@ -53,10 +53,29 @@ impl From<MatchError> for EvaluateError {
     }
 }
 
+// metadata related rules can only be applied to accounts
+pub fn validate_rule_for_specifier(
+    rule: &RequestPolicyRule,
+    specifier: &RequestSpecifier,
+) -> ModelValidatorResult<RequestPolicyError> {
+    match (rule, specifier) {
+        (RequestPolicyRule::AllowListed, RequestSpecifier::Transfer(_))
+        | (RequestPolicyRule::AllowListedByMetadata(_), RequestSpecifier::Transfer(_)) => Ok(()),
+        (RequestPolicyRule::AllowListed, _) | (RequestPolicyRule::AllowListedByMetadata(_), _) => {
+            Err(RequestPolicyError::InvalidRuleForSpecifier {
+                rule: rule.to_string(),
+                specifier: specifier.to_string(),
+            })
+        }
+        (_, _) => Ok(()),
```

Comment:
The cases `AnyOf : vec RequestPolicyRule`, `AllOf : vec RequestPolicyRule`, and `Not : RequestPolicyRule` could contain nested metadata rules that we missed here.

Reply:
Hence, we need to do a full traversal of `RequestPolicyRule`.

Reply:
good catch!

Reply:
Addressed in [f3e14ab](https://github.com/dfinity/orbit/pull/570/commits/f3e14ab3f9430db8007f4bfd2a4f4df36703f3a6)